### PR TITLE
GLA Request Review - Tracking events

### PR DESCRIPTION
**Pull request by puntope**
### Changes proposed in this Pull Request:

Closes part of #1165 

This PR will add support for tracking events in GLA Request Review Module

- Open the modal on Request Review click. `gla_modal_open`
- Close the modal. `gla_modal_closed`  
- Check/Uncheck the Issues solved checkbox.  `gla_request_review_issues_solved_checkbox_click` 
- Expand/Collapse the issue list. `gla_request_review_issue_list_toggle_click`
- Click on the Request Review button inside the modal. `gla_request_review`

### Screenshots:

https://user-images.githubusercontent.com/5908855/155144219-08bbc493-4a5d-4f5e-b816-11a95e13b4cb.mov

### Instructions before the tests

1. Be sure you have an account connected and at least one issue. You can create one issue in the `wp_gla_merchant_issues` table. 
2. Enable tracking in Woo -> Settings -> Advanced -> Woocommere.com -> Enable tracking
Screenshot 2022-02-22 at 15 34 05

3. Open your console and execute this `localStorage.setItem('debug', 'wc-admin*')`
4. Enable verbose mode in console. 
Screenshot 2022-02-22 at 15 35 02
5. Go to WooCommerce -> GLA -> Product feed `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed` 

### Instructions for Events

**Modal Open**

1. Click on the Request Review button. 
2. See how an event `gla_modal_open` appears with the `context` prop as `request_review` 

**Issue expand/collapse**
1. After opening the modal, click on `+1 more issue(s)`. 
2. See how an event `gla_request_review_issue_list_toggle_click` appears with the `action` prop as `expand` 
3. Click now again in the same button, now named `Show less`
4. See how an event `gla_request_review_issue_list_toggle_click` appears with the `action` prop as `collapse` 

**issue solved Checkbox**
1. After opening the modal, click on `I have resolved all the issue(s) listed above.` 
2. See how an event `gla_request_review_issues_solved_checkbox_click` appears with the `action` prop as `check`
3. Click again on the checkbox for disabling it
4. See how an event `gla_request_review_issues_solved_checkbox_click` appears with the `action` prop as `uncheck`

**Close modal**
1. After opening the modal, click on cancel
2. See how an event `gla_modal_closed` appears with the `context` prop as `request_review` and `action` as `cancel`
3. After opening the modal again, click on the X in the corner
4. See how an event `gla_modal_closed` appears with the `context` prop as `request_review` and `action` as `dismiss`
5. After opening the modal again, click outside the modal
6. See how an event `gla_modal_closed` appears with the `context` prop as `request_review` and `action` as `dismiss`

**Review request**
1. After opening the modal, click o. the checkbox and then on the Button Request Review
2. See how an event `gla_modal_closed` appears with the `context` prop as `request_review` and `action` as `request_review`
3. See how an event `gla_request_review` appears 

### Changelog entry

> Add Tracking in Request Review Module

### CODE REVIEW

File: js/src/product-feed/review-request/index.js [hunk 1]
```diff
@@ -29,18 +30,36 @@ const ReviewRequest = ( { account = {} } ) => {
 		return null;
 	}
 
+	const handleNoticeClick = () => {
+		setModalActive( true );
+		recordEvent( 'gla_modal_open', { context: 'request_review' } );
```
Comment (eason9487) [inline js/src/product-feed/review-request/index.js]: 💅 What about giving a constant variable for `'request_review'`?

File: js/src/product-feed/review-request/index.js [hunk 2]
```diff
@@ -29,18 +30,36 @@ const ReviewRequest = ( { account = {} } ) => {
...
+	const handleModalClose = ( action ) => {
+		setModalActive( false );
+		recordEvent( 'gla_modal_closed', {
+			context: 'request_review',
+			action,
+		} );
+	};
+
+	const handleReviewRequest = () => {
+		handleModalClose( 'request_review' );
```
Comment (eason9487) [inline js/src/product-feed/review-request/index.js]: 💅 What about using another `action` string to make it a little differentiated from `context`? Such as `confirm` or `confirm_request_review`.

image

**Review by eason9487 — COMMENTED** [review]
All tracking events are triggered as expected. Left some minor questions. And I think it would be better to add these new tracking names to *src/Tracking/README.md*:
- request_review
- request_review_issue_list_toggle_click
- request_review_issues_solved_checkbox_click

**Comment by puntope** [comment]
Howdy @eason9487 

I pushed some commits based on your suggestions:

- Updated Tracking README. (Added the new functions and fixed some typos in other 2, also my IDE changed the format of some spaces)
- Adding some constant regarding `request-review` and also moving `product-feed/constants.js` file (with only 1 constant) inside `src/constants.js `
- Change `cancel` action in Modal Close to `maybe-later` to be consistent with the Modal Close current definition
- Change `request-review` action innModal Close to `confirm-request-review` 

Could you take another look? 🙇

**Review by eason9487 — APPROVED** [review]
Thanks for the adjustments! LGTM. A little thing about formatting could be considered below. Approved in advance. 👍 

> my IDE changed the format of some spaces

The file was auto-formatted by prettier in #876. So you could run it again to make them have the same spaces in this repo.
```bash
./node_modules/prettier/bin-prettier.js --write ./src/Tracking/README.md
```

**Pull request merged**